### ISSUE TITLE
StackExchange.Redis.RedisTimeoutException: Timeout performing SET mykey

### ISSUE THREAD

**Issue by LILkillaBEE** [StackExchange/StackExchange.Redis#901]
Hi,

I'm having Timeout issues with one of the asp.net core applications with high number of requests.
Application is having issues if it's running on .net core runtime, but on the full .net it runs without exceptions.

I get the timeout exceptions with `netcoreapp2.1` and `netcoreapp1.1`, but `net47` is working without exceptions.

Here is sample code to reproduce the behavior (free redis instance connection string, but the same behavior is with the production redis):
```
        static void Main(string[] args)
        {
            ConnectionMultiplexer redis = ConnectionMultiplexer.Connect(CONFIG_STRING);
            IDatabase db = redis.GetDatabase();

            var cacheTasks = new Task<bool>[15];
            for (int i = 0; i < cacheTasks.Length; i++)
            {
                var tmp = i;
                cacheTasks[i] = Task.Run(() => RedisTest(db, "Nikola", tmp));
            }

            Console.WriteLine("All Tasks Fired");
            Task.WaitAll(cacheTasks);
            Console.WriteLine("All Tasks Completed");
            Console.WriteLine($"Success rate: {cacheTasks.Count(t => t.Result)}/{cacheTasks.Length}");

            Console.WriteLine("\n\n\n");

            Console.ReadLine();
        }

        private static bool RedisTest(IDatabase db, string value, int count)
        {
            var success = true;
            try
            {
                db.StringSet("mykey", value);
                var valueRead = db.StringGet("mykey");
                Console.WriteLine($"{count}. " + (value.Equals(valueRead) ? "success" : "fail"));
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.ToString());
                success = false;
            }

            return success;
        }
```

Multiple threads (`Task.Run`) are intentional and needed in my project.
Am I doing something wrong?

Thank you,
Nikola

(edited by admin: remove private data)

**Comment by mgravell** [comment]
Hi! We are aware that the 1.* codebase was a little less reliable in two scenarios:

- TLS
- .NET Core

You have the combination of the two, so I *expect* that the problems you're seeing are part of that. The good news is that we have invested *huge* amounts of effort into fixing this; the set of changes required for this can *essentially* be paraphrased as "that's 2.0".

It is very much my expectation that 2.0 will completely address this. So; I'm going to close this as [see 2.0](https://github.com/StackExchange/StackExchange.Redis/issues/871) - basically, "yes, we accept this is a problem, and we've done tons of work to make it better - called: 2.0"

As a side note: please be careful with credentials in bug reports; I've redacted that.

**Comment by LILkillaBEE** [comment]
@mgravell thank you for the quick reply.

For the credentials... It was intentionally, temporary redis instance for you guys to test it as easy as possible.
I can confirm that 2.0+ solves this issue.
The thing is that I'm using Microsoft.Extensions.Caching.Redis package and I can't switch to 2.0+ since it's alpha.

What would be your suggestion, use latest 2.0 alpha and implement IDistributed cache by myself?

Thank you,
Nikola

**Comment by mgravell** [comment]
we're not going to have a 1.* fix available in any timescale - our analysis is that it needs significant internal changes, which is what 2.0 *is*. The ASP.NET MVC folks know about our plans, and have their own discussions in place about when/how they update to 2.0.

My focus right now is on getting 2.0 stable and shipped. which we will do as soon as we're happy with it. I would not recommend using it "for real" before it is shipped as 2.0, and note that the nuget version may be significantly behind the repo code.
